migrate/snapshot.go

```diff
@@ -157,10 +180,11 @@ func mangleRoot(n *node) *node {
 	}
 	newRoot.Children["1"] = n
 	etcd := n.Children["_etcd"]
-	delete(n.Children, "_etcd")
+	newEtcd := deepCopyNode(etcd, newRoot)
```

Comment:
what if we just `deepCopyNode(etcd.Children["machines"])`?
i think it is good as long as jed can migrate discovery record.